app/assets/javascripts/initializers/initScrolling.js

```diff
@@ -394,7 +394,9 @@ function fetchNextPageIfNearBottom() {
       fetchNextFollowingPage(indexContainer);
     };
   } else {
-    scrollableElemId = 'articles-list';
+    const useMainContentId = document.getElementById('main-content') !== null;
```

Comment:
refactor (suggestion): Avoid getting a reference to the element with the `"main-content"` ID twice. You could rework that whole function like so.

```diff
function fetchNextPageIfNearBottom() {
  var indexContainer = document.getElementById('index-container');
  var elCheck = indexContainer && !document.getElementById('query-wrapper');
  if (!elCheck) {
    return;
  }

  var indexWhich = indexContainer.dataset.which;

  var fetchCallback;
- var scrollableElemId;
+ var scrollableElem;

  if (indexWhich === 'podcast-episodes') {
-   scrollableElemId = 'articles-list';
+	scrollableElem = document.getElementById('articles-list');
    fetchCallback = function fetch() {
      fetchNextPodcastPage(indexContainer);
    };
  } else if (indexWhich === 'videos') {
-   scrollableElemId = 'video-collection';
+	scrollableElem = document.getElementById('video-collection');
    fetchCallback = function fetch() {
      fetchNextVideoPage(indexContainer);
    };
  } else if (indexWhich === 'followers') {
-   scrollableElemId = 'user-dashboard';
+	scrollableElem = document.getElementById('user-dashboard');
    fetchCallback = function fetch() {
      fetchNextFollowersPage(indexContainer);
    };
  } else if (indexWhich === 'following') {
-   scrollableElemId = 'user-dashboard';
+	scrollableElem = document.getElementById('user-dashboard');
    fetchCallback = function fetch() {
      fetchNextFollowingPage(indexContainer);
    };
  } else {
-   const useMainContentId = document.getElementById('main-content') !== null;
-   scrollableElemId = useMainContentId ? 'main-content' : 'articles-list';
+   scrollableElem =
+     document.getElementById('main-content') ||
+     document.getElementById('articles-list');

    fetchCallback = function fetch() {
      paginate(
        indexContainer.dataset.tag,
        indexContainer.dataset.params,
        indexContainer.dataset.requiresApproval,
      );
    };
  }

  if (
    !done &&
    !fetching &&
    window.scrollY > scrollableElem.scrollHeight - 3700
  ) {
    fetching = true;
    fetchCallback();
  }
}
```

Reply:
yeah I like that more - nice one!